## Writing/Technology Assignment One

### Exploring Matrix Multiplication

##### MATH 254 – Linear Algebra

##### Timothy Paylor

------------------------------
<br>

***Assignment Intrsuctions:***

<br>

***The table [below] shows the first two test scores for Anna, Bruce, Chris, and David. Use the
table to create a matrix M to represent the data. Input M into a software program or
a graphing utility and use it to answer the following questions.$^{1}$***

| Name  | Test 1 | Test 2 |
| ----- | ------ | ------ |
| Anna  | 84     | 96     |
| Bruce | 56     | 72     |
| Chris | 78     | 83     |
| David | 82     | 91     |

In [1]:
# Defining above table as matrix M

import numpy as np

M = np.array([[84, 96], 
              [56, 72], 
              [78, 83], 
              [82, 91]])

print(M)

[[84 96]
 [56 72]
 [78 83]
 [82 91]]


***1. Which test was more difficult? Which was easier? Explain.$^{1}$***

>*Answer:*
<br>
>Test 1 appears to have been more difficult because the average score on Test 1 was 75 and the average score on Test 2 was 85.5. Additionaly, each student individually got a higher score on Test 2 than Test 1.

***2. How would you rank the performances of the four students?$^{1}$***

>*Answer:*
<br>
>The ranking of the students is consistent across both tests with Anna receiving the highest scores, followed by David then Chris then Bruce. Anna's average score across both tests was 90, David's was 86.5, Chris's was 80.5, and Bruce had an average score of 64.

***3. Describe the meanings of the matrix products M*[[1], [0]] and M*[[0], [1]].$^{1}$***

In [2]:
# For M * [[1], [0]]

m = np.array([[1], 
              [0]])

A = M.dot(m)

print (A)

[[84]
 [56]
 [78]
 [82]]


>*Answer:*
<br>
>M[[1], [0]] gives us the Test 1 scores for all four students.

In [3]:
# For M * [[0], [1]]

m = np.array([[0], 
              [1]])

A = M.dot(m)

print (A)

[[96]
 [72]
 [83]
 [91]]


>*Answer:*
<br>
>M[[0], [1]] gives us the Test 2 scores for all four students.

***4. Describe the meanings of the matrix products [1, 0, 0, 0]*M and [0, 0, 1, 0]*M.$^{1}$***

In [4]:
# For [1, 0, 0, 0] * M

m = np.array([1, 0, 0, 0])

A = m.dot(M)

print (A)

[84 96]


>*Answer:*
<br>
>[1, 0, 0, 0]M gives us Anna's scores on both Test 1 and Test 2.

In [5]:
# For [0, 0, 1, 0] * M

m = np.matrix([0, 0, 1, 0])

A = m.dot(M)

print (A)

[[78 83]]


>*Answer:*
<br>
>[0, 0, 1, 0]M gives us Chris' scores on both Test 1 and Test 2.

***5. Describe the meanings of the matrix products M*[[1], [1]] and (0.5)*M*[[1], [1]].$^{1}$***

In [6]:
# For M * [[1], [1]]

m = np.array([[1], 
              [1]])

A = M.dot(m)

print (A)

[[180]
 [128]
 [161]
 [173]]


>*Answer:*
<br>
>M[[1], [1]] returns each student's combined scores from both tests.

In [7]:
# For 0.5 * M * [[1], [1]]

scaled_M = 0.5 * M

m = np.matrix([[1], 
               [1]])

A = scaled_M.dot(m)

print (A)

[[90. ]
 [64. ]
 [80.5]
 [86.5]]


>*Answer:*
<br>
>0.5M[[1], [1]] gives each student's average score across both tests.

***6. Describe the meanings of the matrix products [1, 1, 1, 1]M and 0.25[1, 1, 1, 1]M.$^{1}$***

In [8]:
# For [1, 1, 1, 1] * M

m = np.array([[1, 1, 1, 1]])

A = m.dot(M)

print (A)

[[300 342]]


>*Answer:*
<br>
>[1, 1, 1, 1]M gives the total combined scores per test.

In [9]:
# For 0.25 * [1, 1, 1, 1] * M

scaled_M = 0.25 * M

m = np.array([[1, 1, 1, 1]])

A = m.dot(scaled_M)

print (A)

[[75.  85.5]]


>*Answer:*
<br>
>0.25[1, 1, 1, 1]M gives the average combined score per test.

***7. Describe the meaning of the matrix product [1, 1, 1, 1]M[[1], [1]].$^{1}$***

In [10]:
# [1, 1, 1, 1] * M * [[1], [1]]

m = np.array([[1, 1, 1, 1]])

A = m.dot(M)

m2 = np.array([[1], 
               [1]])

B = A.dot(m2)

print(B)

[[642]]


>*Answer:*
<br>
>[1, 1, 1, 1]*M*[[1], [1]] gives the total points for both tests from all four students.

***8. Use matrix multiplication to find the combined overall average score on
both tests.$^{1}$***

In [11]:
# 0.125 * [1, 1, 1, 1] * M * [[1], [1]]

m = np.array([[1, 1, 1, 1]])

scaled_m = 0.125 * m

A = scaled_m.dot(M)

m2 = np.array([[1], 
               [1]])

B = A.dot(m2)

print(B)

[[80.25]]


>*Answer:*
<br>
>0.125[1, 1, 1, 1]M[[1], [1]] gives the combined overall average score on both tests. The combined overall average score is 80.25.

***9. How could you use matrix multiplication to scale the scores on test 1 by a
factor of 1.1?$^{1}$***

In [12]:
# 1.1 * M * [[1], [0]]

scaled_M = 1.1 * M

m2 = np.array([[1], 
               [0]])

B = scaled_M.dot(m2)

print(B)

[[92.4]
 [61.6]
 [85.8]
 [90.2]]


>*Answer:*
<br>
>1.1*M*[[1], [0]] gives us a scaled (by 1.1) version of the Test 1 scores.

---------------------------
### Sources

<br>

1. Elementary Linear Algebra, 7th Edition, Ron Larson, (c) 2013